lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -310,6 +311,41 @@ sub settings {
     $self->render('test/settings');
 }
 
+sub show_scheduler {
+    my ($self) = @_;
+    use Data::Dumper;
+    #print Dumper($self->stash);
+    my $job          = $self->_stash_job or return $self->reply->not_found;
+    my $module = $self->stash('scheduler_path');
+
+    # if (my $casedir = $job->settings->single({key => 'CASEDIR'})) {
+    #     my $casedir_url = Mojo::URL->new($casedir->value);
+    #     # if CASEDIR points to a remote location let's assume it is a git repo
+    #     # that we can reference like gitlab/github
+    #     last unless $casedir_url->scheme;
+    #     my $refspec = $casedir_url->fragment;
+    #     # try to read vars.json from resultdir and replace branch by actual git hash if possible
+    #     eval {
+    #         my $vars_json = Mojo::File->new($job->result_dir(), 'vars.json')->slurp;
+    #         my $vars      = decode_json($vars_json);
+    #         $refspec = $vars->{TEST_GIT_HASH};
+    #     };
+    #     my $module_path = '/blob/' . $refspec . '/' . $module->script;
+    #     # github treats '.git' as optional extension which needs to be stripped
+    #     $casedir_url->path($casedir_url->path =~ s/\.git//r . $module_path);
+    #     $casedir_url->fragment('');
+    #     return $self->redirect_to($casedir_url);
+    # }
+    my $testcasedir = testcasedir($job->DISTRI, $job->VERSION);
+    my $schedulerpath  = "$testcasedir/schedule/" . $module;
```

Comment:
I'm wondering whether this needs some safety measures to avoid reading files from arbitrary locations (`$module` is user-provided).

Reply:
[Mojo::File](https://mojolicious.org/perldoc/Mojo/File#dirname) would allow you to verify that the parent is what you expect it to be. :thinking:

Reply:
this is a good point. i was thinking if this get limited provided a regex in the routing or even here.